components/governance/org.wso2.carbon.governance.lcm/src/main/java/org/wso2/carbon/governance/lcm/tasks/LCNotificationScheduler.java

```diff
@@ -0,0 +1,323 @@
+    private final String notApplicable = "na";
+
+    /**
+     * This method is used to send notifications for checkpoints in lifecycle. This method is called using the
+     * scheduler task CheckpointNotificationSchedulerTask.
+     */
+    public void run() {
+        try {
+            ArrayList<LCNotification> notifications = getValidNotifications();
+
+            // Send notifications if exists.
+            if (notifications.size() > 0) {
+                // Parse this for a another method.
+                for (LCNotification schedulerBean : notifications) {
+                    // Creating a Checkpoint notification event.
+                    String notificationMessage = getNotificationMessage(schedulerBean);
+                    LifecycleNotificationEvent<String> notificationEvent
+                            = new LifecycleNotificationEvent<>(notificationMessage);
+                    notificationEvent.setTenantId(superTenantId);
+                    notificationEvent.setResourcePath(schedulerBean.getRegPath());
+                    // Sending notification using notification service.
+                    NotificationService notificationService = new EventingServiceImpl();
+                    try {
+                        notificationService.notify(notificationEvent);
+                        if (log.isDebugEnabled()) {
+                            log.debug("Notification '" + notificationMessage + "' sent to notification service.");
+                        }
+                        // Exception is caught because notificationService.notify method throws the exception from Exception
+                        // class. This will be refactored after fixing this JIRA: https://wso2.org/jira/browse/REGISTRY-2433
+                    } catch (Exception e) {
+                        log.error("Error getting registry while getting scheduler objects to send notifications", e);
+                    }
+                }
+            }
+        } catch (RegistryException e) {
+            log.error("Error while getting valid notifications on " + getCurrentDate(), e);
+        }
+    }
+
+    /**
+     * This method is used to add checkpoint notification schedulers to the database. These schedulers are used to
+     * send notifications when they reaches the checkpoints defined in the attached lifecycle.
+     *
+     * @param resource              resource which the lifecycle is attached.
+     * @param lifecycleName         lifecycle name.
+     * @param lifecycleState        new lifecycle state after the state changed.
+     * @param isInvokeAspect        is this add scheduler method is called when invoking a lifecycle.
+     * @throws GovernanceException  Throws when an error occurred while adding a lifecycle notification scheduler to
+     *                              the database.
+     */
+    public void addScheduler(ResourceImpl resource, String lifecycleName, String lifecycleState, boolean isInvokeAspect)
```

Comment:
Hi,

Validated for Illegal arguments in pull request [1], commit[2].
[1] - https://github.com/wso2/carbon-governance/pull/44
[2] - https://github.com/prasa7/carbon-governance/commit/c681dda86c2df780a044e475a55cd5a4b2f5ef89

Thanks